app/src/org/commcare/views/dialogs/CommCareAlertDialog.java

```diff
@@ -47,13 +51,16 @@ public boolean isCancelable() {
         return isCancelable;
     }
 
+    public void makeCancelable() {
+        isCancelable = true;
+    }
+
     public void setOnCancelListener(DialogInterface.OnCancelListener listener) {
         isCancelable = true;
         cancelListener = listener;
     }
 
     public void setOnDismissListener(DialogInterface.OnDismissListener listener) {
-        dialog.setOnDismissListener(listener);
```

Comment:
having this here causes a crash in some instances, since in `CustomViewAlertDialog` the dialog is not initialized until we show it; changed to follow the same pattern as `setOnCancelListener` where we just set the class field here, and then assign it to the dialog in `finalizeView()`